From 6eeca68d65795783243d3634627b4ac8f79e3d15 Mon Sep 17 00:00:00 2001
From: Aditya Kumar Singh <quic_adisi@quicinc.com>
Date: Thu, 28 Mar 2024 23:46:45 +0530
Subject: [PATCH 015/104] hostapd: MLO: skip assoc link processing in ML info

Currently during processing ML info in association request, all links are
iterated over. However, the assoc link info will not be present in the
ML info hence following print is observed during ML association (assoc link
is 1) -

MLD: No link match for link_id=1

Add changes to skip processing for the assoc link. No functionality
changes.

Signed-off-by: Aditya Kumar Singh <quic_adisi@quicinc.com>
---
 src/ap/ieee802_11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 9f7e9afdd..39c63f29b 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -4586,7 +4586,7 @@ int hostapd_process_assoc_ml_info(struct hostapd_data *hapd,
 		struct mld_link_info *link = &sta->mld_info.links[i];
 		bool link_bss_found = false;
 
-		if (!link->valid)
+		if (!link->valid || i == sta->mld_assoc_link_id)
 			continue;
 
 		for_each_mld_link(bss, hapd) {
-- 
2.39.2

